contributed_definitions/nyaml/SPM/NXspm.yaml

```diff
@@ -0,0 +1,832 @@
+        scan_name:
+          exists: optional
+          doc: |
+            The name of the scan, mainly lab defined, that helps to differentiate among different
+            scans performed in lab.
+        current_sensor(NXsensor):
+          exists: optional
+          doc: |
+            This is a link to the current_sensor under the instrument group:
+            entry/experiment_instrument/current_sensor.
+        volatage_sensor(NXsensor):
+          exists: optional
+          doc: |
+            This is a link to the voltage_sensor under the instrument group:
+            entry/experiment_instrument/voltage_sensor.
+        piazo_sensor(NXsensor):
+          exists: optional
+          doc: |
+            This is a link to the piazo_sensor under the instrument group:
+            entry/experiment_instrument/piazo_sensor.
+        (NXscan_control):
+          doc: |
+            The scan control information like scan region or phase space, type of scan (e.g.
+            mesh, spiral, etc.), and scan speed, etc. This group mainly stores the raw scan
+            data. For processed data or final experimental data would go to NXdata group.
+          scan_name:
+            exists: optional
+            doc: |
+              The name of the scan, mainly lab defined, that helps to differentiate among different
+              scans performed in lab. This field is intended for several types of scan control
+              run under the same environment.
+      current_sensor(NXsensor):
+        exists: optional
+        doc: |
+          Information for current sensor.
+        (NXamplifier):
+          exists: optional
+          doc: |
+            An amplifier information for the input signal (e.g. from tip).
+      voltage_sensor(NXsensor):
+        exists: optional
+        doc: |
+          The sensor information for the voltage device.
+        (NXamplifier):
+          exists: optional
+      piazo_sensor(NXsensor):
+        doc: |
+          The piezo sensor refers to the height (or Z) piezo sensor if nothing is
+          mentioned along the X and Y directions.
+        x(NX_NUMBER):
+          unit: NX_LENGTH
+          doc: |
+            The x position of the piezo. In STS experiment, the piezo stays fixed at
+            x,y and z and the the tunneling current is measured with respect to the
+            bias voltage (sweep voltage).
+        y(NX_NUMBER):
+          unit: NX_LENGTH
+          doc: |
+            The y position of the piezo.
+        z(NX_NUMBER):
+          unit: NX_LENGTH
+          doc: |
+            The z position of the piezo.
+        piezo_configuration(NXpiezo_config_spm):
+          exists: optional
+          doc: |
+            The piezo configuration information like piezoelectric calibration and material
+            properties.
+          calibration(NXcalibration):
+            exists: optional
+            doc: |
+              Calibration of the piezo device.
```

Comment:
we already have "calibration" in NXpiezo_config_spm, do we need to redefine it here?

Reply:
To make the app def user-friendly to read the app def. I mentioned some of the fields and groups, which might be used in general.